arrow-json/src/reader.rs

```diff
@@ -582,6 +582,7 @@ where
 /// [`RawDecoder`]: crate::raw::RawDecoder
 /// [#3610]: https://github.com/apache/arrow-rs/issues/3610
 #[derive(Debug)]
+#[deprecated(note = "Use RawDecoder instead")]
```

Comment:
Still need the comment? Actually it reads unclear and seems strange to put issue link there.

```suggestion
#[derive(Debug)]
#[deprecated(note = "Use RawDecoder instead")]
```

Reply:
These are link shorthand, they don't get rendered, but are used to attach links to the text above.

![image](https://user-images.githubusercontent.com/1781103/220367956-c6630641-4a44-4480-a23c-66c2c03eafa0.png)